### PR TITLE
Highlighting

### PR DESCRIPTION
**Pull request by NeilFraser**
Allows multiple blocks to be highlighted when executing multithreaded programs.  No more collision between selections and executions.

### CODE REVIEW

**Comment by NeilFraser** [comment]
@tmickel Could you check this PR out.  It should simplify Scratch's glow functions.

**Comment by tmickel** [comment]
@NeilFraser Seems reasonable to me, we could reasonably replace our `glowBlock` with this version of `highlightBlock`. For us still makes sense to maintain a `glowStack` (or rename it to `highlightStack`), I think. Thanks for the note about the unique IDs - didn't realize that's what it was for.

File: core/inject.js [hunk 1]
```diff
@@ -104,19 +104,22 @@ Blockly.createDom_ = function(container, options) {
   </defs>
   */
   var defs = Blockly.createSvgElement('defs', {}, svg);
+  // Each filter/pattern needs a unique ID for the case of multiple Blockly
+  // instances on a page.  Brower behaviour becomes undefined otherwise.
```
Comment (rachel-fenichel) [inline core/inject.js]: nit spelling: brower -> browser

File: core/workspace_svg.js [hunk 2]
```diff
@@ -65,6 +66,12 @@ Blockly.WorkspaceSvg = function(options) {
    * @const
    */
   this.SOUNDS_ = Object.create(null);
+  /**
+   * List of currently highlighted blocks.
+   * @type !Array.<!Blockly.BlockSvg>
+   * @private
+   */
+  this.highlighted_ = [];
```
Comment (rachel-fenichel) [inline core/workspace_svg.js]: I'd prefer highlightedBlocks_.  Not a very strong preference though.

File: core/workspace_svg.js [hunk 3]
```diff
@@ -537,53 +544,42 @@ Blockly.WorkspaceSvg.prototype.render = function() {
 };
 
 /**
- * Turn the visual trace functionality on or off.
- * @param {boolean} armed True if the trace should be on.
+ * Was used back when block highlighting (for execution) and block selection
```
Comment (rachel-fenichel) [inline core/workspace_svg.js]: Do we know if anyone else is using this?  If so, the console.warn should probably also say to use highlight directly to highlight multiple blocks during execution.

File: demos/interpreter/index.html [hunk 4]
```diff
@@ -183,6 +182,7 @@
         if (!ok) {
           // Program complete, no more code to execute.
           document.getElementById('stepButton').disabled = 'disabled';
+          workspace.highlightBlock(null);
```
Comment (rachel-fenichel) [inline demos/interpreter/index.html]: Does this unhighlight all blocks?

If so, the doc should highlightBlock should call out that highlightBlock(null) will unhighlight everything.

File: core/workspace_svg.js (hunk 3 above)
Comment (NeilFraser) [inline core/workspace_svg.js]: Existing users already need to call highlight directly.  The call to traceOn was designed to deal with the UI collision of using selection as an execution marker.  Now that they are separate, this call does nothing and can be deleted.  There is no API change for existing users.

**Pull request merged**